python-backend/main.py

```diff
@@ -69,33 +54,63 @@ async def analyze(request: Request):
 
     try:
         data = Input(**(await request.json()))
-        (expression, a, b, symbol) = parse(data)
-        pcf = ramanujan.pcf.PCF(str(a), str(b))
-        limit = mpmath.mpf(pcf.limit(depth=data.i))
-        logger.debug(f"PCF limit() returned: {limit}")
-        computed_values = db.identify(values=[limit], wide_search=True)
-        for m in computed_values:
-            logger.debug(f"identify returned: {m}")
+        (a_func, a, b_func, b, symbol) = parse(data)
+
+        if len(data.symbol) > 0:
+            is_convergent = assess_convergence(laurent(a, b, symbol), symbol)
+            # short circuit if the preliminary test fails
+            if is_convergent is False:
+                response = JSONResponse(content={"is_convergent": is_convergent}, status_code=200)
+                return response
+
         # per the link below, "when bi = 1 (the partial numerator) for all i the expression is called a simple continued
         # fraction"
         # b here is the partial numerator (a and b are often used interchangeably which may lead to confusion when there
-        # is no associated image of the fraction)
+        # is no associated image of the continued fraction)
         # https://en.wikipedia.org/wiki/Continued_fraction#Basic_formula
         if data.b == "1":
-            (values, denom_values) = simple_computed_values(a, symbol)
+            (values, num_values, denom_values) = simple_computed_values(a_func, iterations=data.i)
         else:
             # generalized continued fractions, where the partial numerator has its own formula and is not equal to 1
             # https://en.wikipedia.org/wiki/Generalized_continued_fraction
-            (values, denom_values) = generalized_computed_values(a, b, symbol)
+            (values, num_values, denom_values) = generalized_computed_values(a_func, b_func, iterations=data.i)
+
+        limit = values[next(reversed(values))]
+        logger.debug(f"last convergent / limit: {limit}")
+
+        computed_values = db.identify(values=[str(limit)], wide_search=True)
+        for m in computed_values:
+            logger.debug(f"identify returned: {m}")
+
+        error = error_coordinates(values, limit)
+        error_log = error_log_coordinates(error, values, limit)
         body = {
-            "expression": json.dumps(str(expression)),
             "limit": json.dumps("Infinity" if type(limit) is Infinity else str(limit)),
-            "error": json.dumps(error_coordinates(values, limit)),
-            "error_deriv": json.dumps(slope_of_error_coordinates(values, limit)),
+            "error": json.dumps(error),
             "delta": json.dumps(delta_coordinates(values, denom_values, limit)),
             "converges_to": json.dumps(str(computed_values[0] if len(computed_values) > 0 else None))
         }
-        logger.debug(f"Response: {body}")
+
+        if data.debug:
```

Comment:
Added several charts for debugging purposes that are hidden unless a flag is set on the React app.